core/templates/dev/head/domain/exploration/exploration-id-validation.service.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service to validate exploration id
+ */
+
+require('domain/summary/exploration-summary-backend-api.service.ts');
+
+angular.module('oppia').factory('ExplorationIdValidationService', [
+  'ExplorationSummaryBackendApiService',
+  function(ExplorationSummaryBackendApiService) {
+    return {
+      isExpIdValid: function(explorationId) {
+        return ExplorationSummaryBackendApiService.
+          loadPublicExplorationSummaries([explorationId]).then(
```

Comment:
This seems a bit off. The ExplorationSummaryModel is a derived model that is only used for search in the library. It can be recomputed at any time. Could you instead use the ExplorationModel as the source of truth?

Reply:
Hi @seanlip, I tried using the readOnlyExplorationBackendApiService but it throws an orange error box if the exp id is invalid. Moreover, the same check is implemented in the navbar using  `ExplorationSummaryBackendApiService` and it is correct because it takes the array of exploration id's as input, So, many exploration id's can be validated at once. So, I think we should use same convention everywhere. So, if we want to not use `ExplorationSummaryBackendApiService` we will have to make a new endpoint, prehaps. Please correct me if I'm wrong about this?
Thanks

Reply:
Let's make a new endpoint for checking existence and use that everywhere. 

/cc @aks681 who I believe needed a similar endpoint for something else as well.

Reply:
Yeah, I was going to create a similar endpoint that validates everything and returns true or false in #8651, thought that might take a while, so maybe this can be merged in the meantime or should this wait until the other PR is merged?

Reply:
Ah -- that's different actually. This one is just to check for validity of exploration IDs. So they are different things, sorry for conflating them.

@nishantwrp I thought about it more and I think that this is similar enough to the collection editor use case that we can use the existing API for the time being. We can look at this again later after collections are deprecated in favour of stories. Sorry for the confusion!

I'll take a full review pass on the PR now.